bgpd/bgp_vty.c

```diff
@@ -7869,17 +7869,153 @@ static void bgp_show_bestpath_json(struct bgp *bgp, json_object *json)
+			json_object_string_add(json_peer,
+					       "lastErrorCodeSubcode",
+					       errorcodesubcode_hexstr);
+			snprintf(errorcodesubcode_str, 255, "%s%s",
+				 code_str, subcode_str);
+			json_object_string_add(json_peer,
+					       "lastNotificationReason",
+					       errorcodesubcode_str);
+			if (peer->last_reset == PEER_DOWN_NOTIFY_RECEIVED
+			    && peer->notify.code == BGP_NOTIFY_CEASE
+			    && (peer->notify.subcode
+				== BGP_NOTIFY_CEASE_ADMIN_SHUTDOWN
+				|| peer->notify.subcode
+				== BGP_NOTIFY_CEASE_ADMIN_RESET)
+			    && peer->notify.length) {
+				char msgbuf[1024];
+				const char *msg_str;
+
+				msg_str = bgp_notify_admin_message(
+					     msgbuf, sizeof(msgbuf),
+					     (uint8_t *)peer->notify.data,
+					     peer->notify.length);
+				if (msg_str)
+					json_object_string_add(
+					   json_peer,
+					   "lastShutdownDescription",
+					   msg_str);
+			}
+
+		} 
+		json_object_string_add(json_peer, "lastResetDueTo",
+				       peer_down_str[(int)peer->last_reset]);
+	} else {
+		if (peer->last_reset == PEER_DOWN_NOTIFY_SEND
+		    || peer->last_reset == PEER_DOWN_NOTIFY_RECEIVED) {
+			code_str = bgp_notify_code_str(peer->notify.code);
+			subcode_str =
+				bgp_notify_subcode_str(peer->notify.code,
+						       peer->notify.subcode);
+			vty_out(vty, "  Notification %s (%s%s)\n",
+				peer->last_reset == PEER_DOWN_NOTIFY_SEND
+				? "sent"
+				: "received",
+				code_str, subcode_str);
+		} else {
+			vty_out(vty, "  %s\n",
+				peer_down_str[(int)peer->last_reset]);
+		}
+	}
+}
+
+static inline bool bgp_has_peer_failed(struct peer *peer, afi_t afi,
+				       safi_t safi)
+{
+	return ((peer->status != Established) ||
+		!peer->afc_recv[afi][safi]);
+}
+
+static void bgp_show_failed_summary(struct vty *vty, struct bgp *bgp,
+				    struct peer *peer, json_object *json_peer,
+				    int max_neighbor_width, bool use_json)
+{
+	char timebuf[BGP_UPTIME_LEN], dn_flag[2];
+	int len;
+
+	if (use_json) {
+		if (peer_dynamic_neighbor(peer)) {
+			json_object_boolean_true_add(json_peer,
+						     "dynamicPeer");
+		}
+
+		if (peer->hostname)
+			json_object_string_add(json_peer, "hostname",
+					       peer->hostname);
+
+		if (peer->domainname)
+			json_object_string_add(json_peer, "domainname",
+					       peer->domainname);
+		json_object_int_add(json_peer, "connectionsEstablished",
+				    peer->established);
+		json_object_int_add(json_peer, "connectionsDropped",
+				    peer->dropped);
+		peer_uptime(peer->uptime, timebuf, BGP_UPTIME_LEN,
+			    use_json, json_peer);
+		if (peer->status == Established)
+			json_object_string_add(json_peer, "lastResetDueTo",
+					       "AFI/SAFI Not Negotiated");
+		else
+			bgp_show_peer_reset(NULL, peer, json_peer, true);
+	} else {
+		dn_flag[1] = '\0';
+		dn_flag[0] = peer_dynamic_neighbor(peer) ? '*' : '\0';
+		if (peer->hostname
+		    && bgp_flag_check(bgp, BGP_FLAG_SHOW_HOSTNAME))
+			len = vty_out(vty, "%s%s(%s)", dn_flag,
```

Comment:
I see json output has an ability to print domainname, can we print it here as well?

Reply:
I did what the current code does. I suspect adding a domain name in the vty output will blow out the neighbor field way more than you want it to and that's why it isn't used in the vty output, just in the JSON output.